src/main/kotlin/com/roadlink/tripservice/infrastructure/persistence/MySQLSectionRepository.kt

```diff
@@ -71,14 +56,58 @@ class MySQLSectionRepository(
             }
     }
 
-    override fun findAllByTripIds(tripIds: List<UUID>): Set<Section> {
-        return entityManager.createQuery(
-            "SELECT s FROM SectionJPAEntity s WHERE s.tripId in :tripIds",
-            SectionJPAEntity::class.java
-        )
-            .setParameter("tripIds", tripIds)
-            .resultList
-            .map { it.toDomain() }
-            .toSet()
+    override fun findAllByTripIds(tripIds: Set<UUID>): Set<Section> {
+        return find(SectionCommandQuery(tripIds = tripIds)).map { it.toDomain() }.toSet()
     }
-}
\ No newline at end of file
+
+    private fun find(commandQuery: SectionCommandQuery): Collection<SectionJPAEntity> {
+        val cb = entityManager.criteriaBuilder
+        val criteriaQuery = cb.createQuery(SectionJPAEntity::class.java)
+        val root = criteriaQuery.from(SectionJPAEntity::class.java)
+
+        val predicates = mutableListOf<Predicate>()
+
+        if (commandQuery.tripIds.isNotEmpty()) {
+            predicates.add(root.get<UUID>("tripId").`in`(commandQuery.tripIds))
+        }
+
+        if (commandQuery.sectionsIds.isNotEmpty()) {
+            predicates.add(root.get<String>("id").`in`(commandQuery.sectionsIds))
+        }
+
+        if (commandQuery.departureLatitude != null) {
+            predicates.add(
+                root.get<TripPointJPAEntity>("departure").get<Double>("latitude").`in`(commandQuery.departureLatitude)
+            )
+        }
+
+        if (commandQuery.departureLongitude != null) {
+            predicates.add(
+                root.get<TripPointJPAEntity>("departure").get<Double>("longitude").`in`(commandQuery.departureLongitude)
```

Comment:
Aca no convendria que sea un equal (=) en lugar de un in ?

Reply:
si tenés in, podés recibir tanto listas como valores independientes.